Deploay causes  ##[error]String can't be parsed. The StringContainingPath parameter should contain a valid Path ending with an '.exe' or '.dll'. Current string [XXXXX.exe]

Your previous version as described below worked perfectly, albeit with a single version in deploying a TopShelf service.
2020-06-30T10:12:38.4191798Z ==============================================================================
2020-06-30T10:12:38.4192540Z Task         : Windows Service Manager
2020-06-30T10:12:38.4193072Z Description  : Deploys a Windows Service to an Agent or a Deployment Group Target using WinRM.
2020-06-30T10:12:38.4193467Z Version      : 4.3.5
2020-06-30T10:12:38.4193750Z Author       : MDSolutions
2020-06-30T10:12:38.4195805Z Help         : [More Information](https://github.com/Dejulia489/WindowsServiceManager)
2020-06-30T10:12:38.4196583Z ==============================================================================

However, the latest version breaks on the same previously working pipeline with

2020-06-30T17:03:42.3311619Z ==============================================================================
2020-06-30T17:03:42.3312088Z Task         : Windows Service Manager
2020-06-30T17:03:42.3312552Z Description  : Deploys a Windows Service to an Agent or a Deployment Group Target using WinRM.
2020-06-30T17:03:42.3312943Z Version      : 4.4.9
2020-06-30T17:03:42.3313218Z Author       : MDSolutions
2020-06-30T17:03:42.3314866Z Help         : [More Information](https://github.com/Dejulia489/WindowsServiceManager)
2020-06-30T17:03:42.3315612Z ==============================================================================
2020-06-30T17:03:44.0322478Z ##[error]String can't be parsed. The StringContainingPath parameter should contain a valid Path ending with an '.exe' or '.dll'. Current string [C:\Program Files\ZZZZZZZ\YYYYYYYY\XXXXXXXX.exe]

ANy ideas?

We experience the exact same error.

`2020-06-30T18:28:43.2354733Z ##[section]Starting: Deploy Windows Service
2020-06-30T18:28:43.2486271Z ==============================================================================
2020-06-30T18:28:43.2486383Z Task         : Windows Service Manager
2020-06-30T18:28:43.2486458Z Description  : Deploys a Windows Service to an Agent or a Deployment Group Target using WinRM.
2020-06-30T18:28:43.2486514Z Version      : 4.4.9
2020-06-30T18:28:43.2486570Z Author       : MDSolutions
2020-06-30T18:28:43.2486639Z Help         : [More Information](https://github.com/Dejulia489/WindowsServiceManager)
2020-06-30T18:28:43.2486737Z ==============================================================================
2020-06-30T18:28:44.9498565Z [SERVERNAME]: Attempting to locate [SERVICENAME$DEFAULT]
2020-06-30T18:28:46.0064480Z ##[error]String can't be parsed. The StringContainingPath parameter should contain a valid Path ending with an '.exe' or '.dll'. Current string ["C:\Program Files (x86)\XXXService\XXXService.exe" -s SERVERNAME --instancename DEFAULT]
2020-06-30T18:28:46.0378787Z ##[section]Finishing: Deploy Windows Service`

--- a/WindowsServiceManager/WindowsServiceManagerV4/DeployWindowsService.ps1
+++ b/WindowsServiceManager/WindowsServiceManagerV4/DeployWindowsService.ps1
@@ -322,7 +322,7 @@ $scriptBlock = {
             $JustParentPath = $false
         )
         # pattern to analyse Service Startup Command 
-        $matchPattern = '( |^)(?<path>([a-zA-Z]):\\([\\\w\/.-]+)(.exe|.dll))|(( "|^")(?<path2>(([a-zA-Z]):\\([\\\w\/. -]+)(.exe|.dll)))(" |"$))'
+        $matchPattern = '( |^)(?<path>([a-zA-Z]):\\([\\\w\/\(\)\[\]{}öäüÖÄÜ°^!§$%&=`´,;@#+._-]+)(.exe|.dll))|(( "|^")(?<path2>(([a-zA-Z]):\\([\\\w\/\(\)\[\]{}öäüÖÄÜ°^!§$%&=`´,;@#+._ -]+)(.exe|.dll)))(" |"$))'
     
         # check if PathName can be processed
         if ($StringContainingPath -notmatch $matchPattern) {

